src/main/java/Duke.java

```diff
@@ -1,10 +1,411 @@
+                System.out.println(LINES + TASK_DONE_MESSAGE + LINES);
+            } else {
+                tasks.get(index - 1).markAsDone();
+                System.out.print(LINES + TASK_COMPLETE_MESSAGE + tasks.get(index - 1)  +
+                        System.lineSeparator() + LINES);
+                updateDatabase();
+            }
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println(LINES + NO_TASK_DONE_NUMBER_MESSAGE + LINES);
+        } catch (NumberFormatException e) {
+            System.out.println(LINES + DONE_NOT_INTEGER_MESSAGE + LINES);
+        }
+    }
+
+    //Adds task to tasks
+    public static void addTask(String input) {
+        if (input.toLowerCase().startsWith("todo")) {
+            addTodo(input);
+            int indexOfAddedTask = tasks.size() - 1;
+            appendDatabase("T", tasks.get(indexOfAddedTask).description, "0");
+        } else if (input.toLowerCase().startsWith("deadline")) {
+            addDeadline(input);
+            int indexOfAddedTask = tasks.size() - 1;
+            Deadline addedDeadline = (Deadline) tasks.get(indexOfAddedTask);
+            String deadlineInput = addedDeadline.description + "/by" + addedDeadline.by;
+            appendDatabase("D", deadlineInput, "0");
+        } else if (input.toLowerCase().startsWith("event")){
+            addEvent(input);
+            int indexOfAddedTask = tasks.size() - 1;
+            Event addedEvent = (Event) tasks.get(indexOfAddedTask);
+            String eventInput = addedEvent.description + "/at" + addedEvent.from;
+            appendDatabase("E", eventInput, "0");
+        }
+    }
+    private static void addTodo(String input) {
+        try {
+            tasks.add(new Todo(input.substring(5)));
+            int taskIndex = tasks.size() - 1;
+            printTaskAcknowledgement(tasks.get(taskIndex));
+        } catch (StringIndexOutOfBoundsException e){
+            System.out.println(LINES + PLEASE_ADD_A_TASK_DESCRIPTION + LINES);
+        }
+    }
+    private static void addDeadline(String input) {
+        try {
+            if (input.trim().length() == 8) {
+                System.out.println(LINES + PLEASE_ADD_A_TASK_DESCRIPTION + LINES);
+            } else if (!input.contains("/by")) {
+                System.out.println(LINES + MISSING_BY_MESSAGE + LINES);
+            } else {
+                int slashIndex = input.indexOf("/by");
+                String task = input.substring(9, slashIndex).trim();
+                String dueDate = input.substring(slashIndex + 3).trim();
+                tasks.add(new Deadline(task, dueDate));
+                int taskIndex = tasks.size() - 1;
+                printTaskAcknowledgement(tasks.get(taskIndex));
+            }
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println(LINES + MISSING_IN_DEADLINE_MESSAGE + LINES);
+        }
+    }
+    private static void addEvent(String input) {
```

Comment:
You may want to add spacing between the end of addDeadline and start of addEvent.